bootstrap/bootstrap.sh

```diff
@@ -28,8 +29,54 @@ sudo apt-get update
 sudo apt-get -y dist-upgrade
 
 #----nodejs install
-curl -sL https://deb.nodesource.com/setup_7.x | sudo -E bash -
+curl -sL https://deb.nodesource.com/setup_6.x | sudo -E bash -
 sudo apt-get install -y nodejs
 
-#----alsa install
-sudo apt-get install -y alsa-base alsa-utils
+#----install official requirements
+sudo apt-get install -y alsa-base alsa-utils libasound2-dev
+
+#----install missing pigpio in Raspbian Lite (the command npm install pigpio will be exec by package.json)
+sudo apt-get install pigpio
+
+#----install git and download tjbot
+if [ ! -d "${TJBOT_FOLDER}/recipes/conversation" ]; then
+    sudo apt-get install -y git
+
+    rm -Rf /home/pi/git/tjbot
+    git clone https://github.com/ibmtjbot/tjbot.git /home/pi/git/tjbot
+
+    TJBOT_FOLDER='/home/pi/git/tjbot'
+fi
+
+#----install conversation (to install it will resolve tjbotlib and other dependencies)
+cd $TJBOT_FOLDER
+cd recipes/conversation
+echo "path: $PWD"
+
+sudo npm install --unsafe-perm -g node-gyp
```

Comment:
I'm not near a Pi at the moment and can't perform any tests. However, if I'm not mistaken, I think the statement from npm above only applies when in a package directory and no package is specified on the command line. In our case here the packages `readline-sync` or `node-gyp` are specified on the command line.

My current understanding is that calling `sudo npm install --unsafe-perm -g readline-sync` in directory `/home/pi/tjbot/bootstrap/tests` will install package `readline-sync` globally in `/usr/lib/node_modules`. It will not install the current package context (`/home/pi/tjbot/bootstrap/tests`) globally.

Calling `sudo npm install --unsafe-perm -g` in directory `home/pi/tjbot/bootstrap/tests` will install the current package context (`/home/pi/tjbot/bootstrap/tests`) globally.

But as already mentioned I'm not near a Pi to try it out.

> We could do more tests without sudo and not global.

This is a good idea. To be honest, I think it's essential.

> Have you tried to used `npm install` in bootstrap.sh using your rpi?

Yes, I tried it yesterday but only for `home/pi/tjbot/recipes/conversation`. I'll try the whole thing again when I'm back at the Pi.

I was just wondering, is it possible that you called `sudo npm install --unsafe-perm -g` in directory `home/pi/tjbot/bootstrap/tests` or `/home/pi/tjbot/recipes/conversation` at some point in the past? If this is the case, the corresponding packages will have been installed globally in the past and have been there ever since. This would explain how tests are still working now.

Note that `bootstrap.sh` only partially cleans up an old installation. It removes local files with `rm -Rf /home/pi/tjbot` but it doesn't uninstall npm packages that are installed globally. This is another reason not to install globally.

And another thought, if users edit/create files in `/home/pi/tjbot` and call `bootstrap.sh` a second time, they may not be too happy that the call to `rm -Rf /home/pi/tjbot` will delete their work.